src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionBasicTests.cs

```diff
@@ -135,6 +135,29 @@ public void ConnectionTimeoutTestWithThread()
             Console.WriteLine($"ConnectionTimeoutTestWithThread: Elapsed Time {theMax} and threshold {threshold}");
         }
 
+        [Fact]
+        public void ExceptionsWithMinPoolSizeCanBeHandled()
+        {
+            int count = 0;
+            int expected = 3;
+            string connectionString = $"Data Source={Guid.NewGuid().ToString()};uid=random;pwd=asd;Connect Timeout=2; Min Pool Size=3";
+            while (count < expected)
+            {
+                using (SqlConnection connection = new SqlConnection(connectionString))
+                {
+                    try
+                    {
+                        connection.Open();
```

Comment:
Is this going to fail every time?  If so, how about using Assert.Throws? That would make it something like:
```C#
string connectionString = ...;
for (int i = 0; i < 3; i++)
{
    using (var connection = new SqlConnection(connectionString))
    {
        Assert.Throws<WhateverExceptionType>(() => connection.Open());
    }
}
```

Reply:
Yes it will fail everytime. I will adapt the testcase.

Reply:
Note: The exception thrown is either InvalidOperationException (if the connection pool timeout is triggered) or SqlException (if the network error in connecting to invalid Data Source happens before the Connection Pool timeout) . So the specific exception type cannot be determined. I will use `Assert.ThrowsAny<Exception>(() => connection.Open());`

Reply:
> So the specific exception type cannot be determined. I will use Assert.ThrowsAny<Exception>(() => connection.Open());

In that case you can do:
```C#
Exception error = Record.Exception(() => connection.Open());
Assert.True(error is InvalidOperationException || error is SqlException, $"Unexpected exception: {error}");
```